### PR TITLE
Add proxy service

### LINKED ISSUE #2: Create proxy service
Please, add all settings for proxy service.

### PR DESCRIPTION
**Pull request by camelya58**
Closes #2

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -21,6 +21,9 @@ dependencies {
 
     /* eureka client */
     implementation 'org.springframework.cloud:spring-cloud-starter-netflix-eureka-client:2.2.2.RELEASE'
+
+    /* zuul */
+    implementation 'org.springframework.cloud:spring-cloud-starter-netflix-zuul:2.2.3.RELEASE'
 }
 
 def profile = "dev"
```
Comment (mamadaliev) [inline build.gradle]: ~~Change the `dev` to `develop`.~~

File: src/main/resources/bootstrap.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  port: 8000
+
+spring:
+  application:
+    name: api-gateway
+
+eureka:
+  client:
+    service-url:
+      defaultZone: http://localhost:8761/eureka
```
Comment (mamadaliev) [inline src/main/resources/bootstrap.yml]: Change the `localhost` to the `registry-service`.

File: src/main/resources/bootstrap.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+
```
Comment (mamadaliev) [inline src/main/resources/bootstrap.yml]: Delete extra lines.

File: src/main/resources/bootstrap.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+      connect-timeout-millis: 10000
+      socket-timeout-millis: 60000
+zuul:
+  routes:
+    client1:
```
Comment (mamadaliev) [inline src/main/resources/bootstrap.yml]: Change the `client1 ` to the `post`.

File: src/main/resources/bootstrap.yml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+      serviceId: post-api
+      path: /api/v1/post/**
+    client2:
```
Comment (mamadaliev) [inline src/main/resources/bootstrap.yml]: Change the `client2 ` to the `account`.

File: src/main/java/com/reckue/gateway/GatewayApplication.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+/**
+ * Class GatewayApplication represents a main class for proxy service.
+ *
+ * @author Kamila Meshcheryakova
+ */
+@EnableZuulProxy
+@EnableEurekaClient
+@SpringBootApplication
+public class GatewayApplication {
+    public static void main(String[] args) {
```
Comment (mamadaliev) [inline src/main/java/com/reckue/gateway/GatewayApplication.java:18]: Add new line before main method.

File: build.gradle (hunk 1 above)
Comment (mamadaliev) [reply build.gradle]: Delete this line, because spring profile is absent.

**Review by mamadaliev — CHANGES_REQUESTED** [review]
Please fix

File: src/main/resources/bootstrap.yml (hunk 2 above)
Comment (camelya58) [reply src/main/resources/bootstrap.yml]: It does't work if change it to "http://registry-service:8761/eureka".
Or i don't understand what you want.

Comment (mamadaliev) [reply src/main/resources/bootstrap.yml]: Then need to add spring profiles.
For develop profile:
```
eureka:
  client:
    service-url:
      defaultZone: http://localhost:8761/eureka
```

Add for cloud profile:
```
eureka:
  client:
    service-url:
      defaultZone: http://registry-service:8761/eureka
```

Comment (mamadaliev) [reply src/main/resources/bootstrap.yml]: On the cloud it should look like this:
```
http://{service-name}:{service-port}/eureka
```

**Review by mamadaliev — APPROVED** [review]
Approve

**Pull request merged**